src/components/tutorial-meta/components/badges/helpers.ts

```diff
@@ -0,0 +1,71 @@
+import React, { ForwardRefExoticComponent, RefAttributes } from 'react'
+import { IconProps } from '@hashicorp/flight-icons/svg-react/types'
+import { IconClock16 } from '@hashicorp/flight-icons/svg-react/clock-16'
+import { IconPlay16 } from '@hashicorp/flight-icons/svg-react/play-16'
+import { IconTerminalScreen16 } from '@hashicorp/flight-icons/svg-react/terminal-screen-16'
+import { EditionOption } from 'lib/learn-client/types'
+import { ProductUsed as ClientProductUsed } from 'lib/learn-client/types'
+import {
+  getBadgeComponent,
+  ProductDisplayOption,
+  BadgeComponentProps,
+} from './components/badge'
+import { BadgeOptions } from '.'
+
+type DisplayOptions = {
+  [K in keyof BadgeOptions]: DisplayOption | ProductDisplayOption[]
+}
+
+type DisplayOption = {
+  label: string
+  icon?: ForwardRefExoticComponent<IconProps & RefAttributes<SVGSVGElement>>
+}
+
+const editionDisplayOptions: { [K in EditionOption]: string } = {
+  [EditionOption.tfcFree]: 'Terraform Cloud',
+  [EditionOption.tfcTeam]: 'Team',
+  [EditionOption.tfcGov]: 'Team & Governance',
+  [EditionOption.enterprise]: 'Enterprise',
+  [EditionOption.tfcBiz]: 'Business',
+  [EditionOption.hcp]: 'HCP',
+  [EditionOption.openSource]: 'Open Source',
+}
+
+// Returns a map of badge display options and a Default Badge component to render
+export function generateBadges(
+  readTime,
+  products,
+  edition
+): [DisplayOptions, React.FC<BadgeComponentProps>] {
+  const displayOptions = {
+    readTime: {
+      label: `${readTime} min`,
+      icon: IconClock16,
+    },
+    isBeta: { label: 'Beta' },
+    edition: { label: editionDisplayOptions[edition] },
+    products: products.map((p) => ({
+      label: p.name,
+      slug: p.slug,
+    })) as ProductDisplayOption[], // There can be many products associated with a single tutorial
+    hasVideo: { label: 'Video', icon: IconPlay16 },
+    isInteractive: { label: 'Interactive', icon: IconTerminalScreen16 },
+  }
+  const DefaultBadgeComponent = getBadgeComponent(displayOptions)
+
+  return [displayOptions, DefaultBadgeComponent]
```

Comment:
I wanted to retain flexibility with badge rendering by returning the display data along with a component, but the view can also use the data to render a custom badge component, similar to the handling of the product badge variant.